examples/blockchain/src/network.rs

```diff
@@ -233,130 +95,200 @@ impl aleph_bft::Network<NetworkData> for Network {
 }
 
 pub struct NetworkManager {
-    swarm: Swarm<Behaviour>,
+    id: usize,
+    address: Address,
+    addresses: Vec<Option<Address>>,
+    bootnodes: HashMap<u32, Address>,
+    listener: TcpListener,
+    consensus_tx: UnboundedSender<NetworkData>,
     consensus_rx: UnboundedReceiver<(NetworkData, Recipient)>,
+    block_tx: UnboundedSender<Block>,
     block_rx: UnboundedReceiver<Block>,
 }
 
-impl Network {
+impl NetworkManager {
     pub async fn new(
-        node_ix: NodeIndex,
+        id: usize,
+        ip_addr: String,
+        n_nodes: usize,
+        bootnodes: HashMap<u32, Address>,
     ) -> Result<
         (
             Self,
-            NetworkManager,
+            Network,
             UnboundedSender<Block>,
             UnboundedReceiver<Block>,
             UnboundedSender<NetworkData>,
             UnboundedReceiver<NetworkData>,
         ),
         Box<dyn Error>,
     > {
-        let local_key = identity::Keypair::generate_ed25519();
-        let local_peer_id = PeerId::from(local_key.public());
-        info!(target: "Blockchain-network", "Local peer id: {:?}", local_peer_id);
-
-        // Create a keypair for authenticated encryption of the transport.
-        let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
-            .into_authentic(&local_key)
-            .expect("Signing libp2p-noise static DH keypair failed.");
-
-        // Create a tokio-based TCP transport use noise for authenticated
-        // encryption and Mplex for multiplexing of substreams on a TCP stream.
-        let transport = TokioTcpConfig::new()
-            .nodelay(true)
-            .upgrade(upgrade::Version::V1)
-            .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
-            .multiplex(mplex::MplexConfig::new())
-            .boxed();
+        let mut addresses = vec![None; n_nodes];
+        for (id, addr) in &bootnodes {
```

Comment:
If you change addresses to `HashMap` you could just add the new address to bootnodes and pass them ass addresses.